homeassistant/components/samsungtv/media_player.py

```diff
@@ -277,7 +285,11 @@ def media_previous_track(self):
     def turn_on(self):
         """Turn the media player on."""
         if self._mac:
-            self._wol.send_magic_packet(self._mac)
+            if self._broadcast:
+                self._wol.send_magic_packet(
```

Comment:
We should remove the whole magic packet PyPI package from this integration and instead use the `wake_on_lan` integration .

Reply:
But that's precisely what `wake_on_lan` [uses](https://github.com/home-assistant/home-assistant/blob/2c07bfb9e0ac03fdf74b83152f4511ed7f7104b0/homeassistant/components/wake_on_lan/switch.py#L73) - what do you mean by using an integration instead?

Reply:
Exactly. DOn't rely on the package inside samsung. Instead, call the wake on lan service offered by the wake on lan integration

Reply:
Thanks for an idea, see new commit is below. However, the test is still referring to wakeonlan library and I haven't figured out a pattern of how to convert existing assert to use integration - a hint will be much appreciated!